### PR TITLE
Test name:  test_superadmin_child_tenant_crud

### PR DESCRIPTION
**Pull request by patchkez**
Test CRUD operations for CFME child tenants, where several levels of tenants are created. 

{{pytest: cfme/tests/configure/test_access_control.py -v -k "test_superadmin_child_tenant_crud"}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 0f8912ef152c4833a5d689bc004fe2168414f3fa:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/configure/test_access_control.py [hunk 1]
```diff
@@ -690,3 +690,48 @@ def _delete_tenant_and_project():
...
+        * Delete all created tenants in reversed order
+    """
+
+    tenant = None
+    tenant_list = []
+
+    @request.addfinalizer
+    def _delete_tenants():
+        # reversed because we need to go from the last one
+        for tenant in reversed(tenant_list):
```
Comment (psav) [inline cfme/tests/configure/test_access_control.py]: Do these have to be deleted in the reverse order?

File: cfme/tests/configure/test_access_control.py [hunk 2]
```diff
@@ -690,3 +690,48 @@ def _delete_tenant_and_project():
...
+    for i in range(1, number_of_childrens + 1):
+        new_tenant = ac.Tenant(
+            name="tenant{}_{}".format(i, fauxfactory.gen_alpha(4)),
+            description=fauxfactory.gen_alphanumeric(16),
+            parent_tenant=tenant)
+        tenant_list.append(new_tenant)
+        new_tenant.create()
+        tenant = new_tenant
+
+    tenant_update = tenant.parent_tenant
```
Comment (psav) [inline cfme/tests/configure/test_access_control.py]: Just confirming this is only supposed to be testing updates of one tenant here

File: cfme/tests/configure/test_access_control.py (hunk 1 above)
Comment (mfalesni) [inline cfme/tests/configure/test_access_control.py]: I was helping Pavol with this and he left so I am writing why - if the children tenants/projects are clean, then it is possible to delete just the root one and that should delete all the child nodes. But if they have something assigned (I don't remember exactly), then you have to resolve it from bottom, so I believe the correct approach is to delete them one by one. And eventually check the mass deleting via the root node in a separate test case :)

Comment (psav) [inline cfme/tests/configure/test_access_control.py]: That's cool I was just checking, not saying that it shouldn't be done this way ;)

Comment (patchkez) [inline cfme/tests/configure/test_access_control.py]: Yes, mfalesni is right, we have to delete tenants in reverse order. When there is group assigned to any child tenant (this is not case for our test, we do not create/assign groups), deletion of top tenant will fail.   
https://bugzilla.redhat.com/show_bug.cgi?id=1292157

I will try to implement check for existing groups/users assigned to tenant in the future. These would be deleted first before deleting tenant.

File: cfme/tests/configure/test_access_control.py (hunk 2 above)
Comment (patchkez) [inline cfme/tests/configure/test_access_control.py]: Yes, correct. We update name/description of last-1 tenant. This is just to test that relationship between last and his parent tenant will not break when renaming parent tenant.

File: cfme/tests/configure/test_access_control.py (hunk 1 above)
Comment (mfalesni) [inline cfme/tests/configure/test_access_control.py]: @patchkez Justy FYI once you implement that, make it optional, so we can make it fail :)

**Pull request merged**